refactor: remove lib/* into extensions

sathvikbhagavan opened the pull request:
Fixes: #513

ChrisRackauckas commented:
Looks fine but the docs need to be updated to not be importing the old libraries.

sathvikbhagavan commented:
@ChrisRackauckas, I have fixed the docs and downgrade CI as well. 
One more thing -  I renamed SurrogateRandomForests to SurrogateXGBoost as it uses XGBoost internally and not RandomForests 😅

ChrisRackauckas commented on Project.toml:
```diff
@@ -11,44 +11,76 @@ GLM = "38e38edf-8417-5370-95a0-9cbb8c7f171a"
+
+[extensions]
+SurrogatesAbstractGPsExt = "AbstractGPs"
+SurrogatesFluxExt = ["Flux", "Optimisers"]
```
Optimisers being necessary will need to be clearly documented.

ChrisRackauckas replied:
```suggestion
SurrogatesFluxExt = ["Flux"]
```

ChrisRackauckas commented on docs/pages.jl:
```diff
@@ -7,7 +7,7 @@ pages = ["index.md"
              "Linear" => "LinearSurrogate.md",
              "InverseDistance" => "InverseDistance.md",
-             "RandomForest" => "randomforest.md",
+             "XGBoost" => "xgboost.md",
```
that's fine, better actually

ChrisRackauckas commented on docs/src/neural.md:
```diff
@@ -1,9 +1,5 @@
 # Neural Network Surrogate Tutorial
 
-!!! note
-    
-    This surrogate requires the 'SurrogatesFlux' module, which can be added by inputting "]add SurrogatesFlux" from the Julia command line.
```
Missing the mention of Optimizers?

sathvikbhagavan commented on ext/SurrogatesFluxExt.jl:
```diff
@@ -1,26 +1,13 @@
 
+using Surrogates: NeuralSurrogate
 using SurrogatesBase
 using Optimisers
```
Hmm, I thought that was deprecated in favour of using Optimisers.jl

sathvikbhagavan commented on ext/SurrogatesFluxExt.jl:
```diff
@@ -1,26 +1,13 @@
 
+using Surrogates: NeuralSurrogate
 using SurrogatesBase
 using Optimisers
```
I think it should be `import Flux: Flux.Optimisers` with a `s` and not `z`

ChrisRackauckas commented on ext/SurrogatesFluxExt.jl:
```diff
@@ -1,26 +1,13 @@
+import: NeuralSurrogate
 using SurrogatesBase
-using Optimisers
+using Flux: Flux.Optimizers
```
```suggestion
using Flux: Flux.Optimisers
```

ChrisRackauckas commented on ext/SurrogatesFluxExt.jl:
```diff
@@ -1,26 +1,13 @@
-module SurrogatesFlux
+module SurrogatesFluxExt
 
+import: NeuralSurrogate
```
I've had cleaner last minute changes.
